### PR TITLE
Add LocalizableResource class for parsing complicated path pattern

### PR DESCRIPTION
**Pull request by oliverhu**
Now support

hdfs://123.txt::abc.txt#archive
SOURCEFILE :: LOCALFILENAME :: TYPE

### CODE REVIEW

**Comment by erwa** [comment]
Can you rebase on `master`? There's a conflict that can't be auto-resolved.

**Comment by oliverhu** [comment]
mm, totally forgot about this PR, will rebase master

File: tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        if (tuple.length == 2) {
+            localFileName = tuple[1];
+        }
+
+        if (sourceFileStatus.isDirectory()) {
+            isDirectory = true;
+        }
+    }
+
+    public LocalResource getLocalResource() {
```
Comment (erwa) [inline tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java]: `toLocalResource()` sounds more accurate to me (you're converting from one representation to another).

File: tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    public LocalizableResource(String path) {
+        this.path = path;
+    }
+
+    public void parse(FileSystem fs) throws ParseException, IOException {
+        String filePath = path;
+        resourceType = LocalResourceType.FILE;
+        if (path.contains(Constants.ARCHIVE_SUFFIX)) {
+            resourceType = LocalResourceType.ARCHIVE;
+            filePath = path.replace(Constants.ARCHIVE_SUFFIX, "");
```
Comment (hungj) [inline tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java]: In addition to @erwa's path.endswith comment, 

path.substring(0, path.length() - Constants.ARCHIVE_SUFFIX.length())

File: tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+    public String getLocalFileName() {
+        return localFileName;
+    }
+
+    public LocalizableResource(String path) {
+        this.path = path;
+    }
+
+    public void parse(FileSystem fs) throws ParseException, IOException {
```
Comment (hungj) [inline tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java]: Any reason this needs to be a separate method? Can we just make this private and call it in the constructor?

File: tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        String filePath = path;
+        resourceType = LocalResourceType.FILE;
+        if (path.contains(Constants.ARCHIVE_SUFFIX)) {
```
[the remark this answers is not in the document]
Comment (oliverhu) [reply tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java]: updated to `endsWith`, i think case-insensitive is fine

File: tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java (hunk 2 above)
Comment (oliverhu) [reply tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java]: ok.. this makes sense to use together with @erwa 's insensitive archive_suffix comment

File: tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java (hunk 3 above)
Comment (oliverhu) [reply tony-core/src/main/java/com/linkedin/tony/LocalizableResource.java]: that's a good point, updated

**Pull request merged**